components/21-atoms/table/table.functions.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+var ResponsiveTable = function (element, options) { // eslint-disable-line no-unused-vars
+  /**
+   * The table wrapper.
+   */
+  var table = null;
+
+  /**
+   * The table caption.
+   */
+  var caption = null;
+
+  /**
+   * The options for the table.
+   */
+  var opts = {};
+
+  /**
+   * Determine if the table should be focusable.
+   */
+  var determineFocusable = function () {
+    var scrollableWidth = table.parentNode.querySelector('table').scrollWidth; // eslint-disable-line no-console
+    var containerWidth = table.parentNode.clientWidth; // eslint-disable-line no-console
+
+    // Check if element is scrollable.
+    if (scrollableWidth <= containerWidth) {
+      // If not remove the tab focus.
+      table.removeAttribute('tabindex');
+    }
+    else {
+      caption.innerText = caption.innerText + ' ' + opts.scrollableText;
+    }
+  };
+
+  /**
+   * Setup the responsive table.
+   */
+  var setupResponsiveTable = function () {
+    addTableConotainerAttributes();
+    addCaptionAttribute();
+    determineFocusable();
+  };
+
+  /**
+   * Add the ID attribute to the caption to reference is in
+   * aria-labelledby on the table container wrapper.
+   */
+  var addCaptionAttribute = function () {
+    if (!caption.hasAttribute('id')) {
+      caption.setAttribute('id', 'caption');
+    }
+  };
+
+  /**
+   * Add attributes to the table container.
+   */
+  var addTableConotainerAttributes = function () {
+    table.setAttribute('tabindex', '0');
+    table.setAttribute('role', 'group');
+    table.setAttribute('aria-labelledby', 'caption');
+  };
+
+  /**
+   */
+  /**
+   * Entry point of the script.
+   *
+   * @param element
+   *   The element.
+   * @param options
+   *   The options for the element.
+   */
+  var init = function (element, options) {
+    if (!element) {
+      return;
+    }
+
+    table = element;
+    caption = table.querySelector('caption');
+    opts = options;
+
+    setupResponsiveTable();
+  };
+
+  init(element, options);
```

Comment:
No need to pass these through as parameters, since they're both available within the scope of the `ResponsiveTable` function.

```suggestion
  init();
```